pipeline/filters/aws-metadata.md

```diff
@@ -49,9 +52,84 @@ $ bin/fluent-bit -c /PATH_TO_CONF_FILE/fluent-bit.conf
     account_id true
     hostname true
     vpc_id true
+    tags_enabled true
 
 [OUTPUT]
     Name stdout
     Match *
 ```
 
+## EC2 Tags
+
+EC2 Tags are a useful feature that enables you to label and organize your EC2 instances by creating custom-defined key-value pairs. These tags are commonly utilized for resource management, cost allocation, and automation. Consequently, including them in the Fluent Bit generated logs is almost essential.
+
+To achieve this, AWS Filter can be configured with `tags_enabled true` to enable the _tagging_ of logs with the relevant EC2 instance tags. This setup ensures that logs are appropriately tagged, making it easier to manage and analyze them based on specific criteria.
+
+### Requirements
+
+For Fluent Bit to retrieve EC2 tags, the [instance-metadata-tags](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-instance-metadata-options.html) option must be enabled on the EC2 instance where it is running. If this option is not enabled, Fluent Bit will not be able to fetch the tags, resulting in its inability to identify them. As a consequence, Fluent Bit will operate as if there are no tags associated with the EC2 instance.
+
+That's it! However, proper configuration is still necessary, and this will be elaborated on in the following section.
+
+### Configuration
+
+As outlined in the 'Configuration Parameters' section, there are currently three options associated with EC2 Tags functionality:
+
+ - `tags_enabled`: This option determines whether the EC2 Tags functionality should be enabled. If it is enabled without any additional configuration parameters specified, all tags will be injected into the logs. On the other hand, if it is set to false, the feature is disabled, and `tags_include` and `tags_exclude` will have no effect.
+ - `tags_include`: This option specifies the particular tags that should be included in the logs.
+ - `tags_exclude`: This option specifies the specific tags that should be excluded from the logs.
+
+Both `tags_include` and `tags_exclude` accept a string that represents a list of tag names separated by a comma. Note that tag names do not support regular expressions or wildcards. The values and tag names must be exactly equal in order to be included or excluded.
+
+Here is an example of how `tags_include` can be used in the configuration:
+```
+[FILTER]
+    Name aws
+    Match *
+    tags_enabled true
+    tags_include department,project
+```
+
+In this example, the `tags_include` parameter is set to `department,project`, which means only tags with the names `department` and `project` will be injected into the logs. Any other tags associated with the EC2 instance will be excluded from the logs.
+
+What happens if we attempt to include and exclude tags at the same time? Fluent Bit will fail if both `tags_include` and `tags_exclude` are defined.
+
```

Comment:
since you have a tags include and tags exclude example below, I think this is repetitive and not needed, user can just read those sections, which have examples of the output which is great